.github/workflows/main.yml

```diff
@@ -0,0 +1,439 @@
+          fail_ci_if_error: true
+          dry_run: true
+
+  aura-teardown:
+    needs: [cla-check, cache-dependencies]
+    if: |
+      always() &&
+      (needs.cla-check.result == 'success' || needs.cla-check.result == 'skipped') &&
+      (needs.cache-dependencies.result == 'success')
+
+    strategy:
+      matrix:
+        aura-instance: [free, professional]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+          cache: yarn
+      - name: Install dependencies
+        run: yarn --immutable
+      - name: Install ts-node and typescript
+        run: npm install -g ts-node typescript
+      - name: Uppercase instance name
+        run: |
+          lowercase_instance=${{ matrix.aura-instance }}
+          echo "AURA_URI_SECRET=AURA_${lowercase_instance^^}_URI" >>${GITHUB_ENV}
+          echo "AURA_PASSWORD_SECRET=AURA_${lowercase_instance^^}_PASSWORD" >>${GITHUB_ENV}
+      - name: Delete all data in instance
+        run: ts-node tests/integration/teardown.ts
+        working-directory: packages/graphql
+        env:
+          NEO_USER: neo4j
+          NEO_PASSWORD: ${{ secrets[env.AURA_PASSWORD_SECRET] }}
+          NEO_URL: ${{ secrets[env.AURA_URI_SECRET] }}
+
+  integration-tests-aura:
+    needs: aura-teardown
+    if: |
+      always() &&
+      (needs.aura-teardown.result == 'success')
+
+    strategy:
+      matrix:
+        package: [graphql, ogm]
+        aura-instance: [free, professional]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+          cache: yarn
+      - name: Install dependencies
+        run: yarn --immutable
+      - name: Uppercase instance name
+        run: |
+          lowercase_instance=${{ matrix.aura-instance }}
+          echo "AURA_URI_SECRET=AURA_${lowercase_instance^^}_URI" >>${GITHUB_ENV}
+          echo "AURA_PASSWORD_SECRET=AURA_${lowercase_instance^^}_PASSWORD" >>${GITHUB_ENV}
+      - name: Run @neo4j/graphql integration tests
+        run: yarn test:int -t '^(?!delete should delete a movie, a single nested actor and another movie they act in|update should delete a nested actor and one of their nested movies, within an update block)' --coverage
+        working-directory: packages/${{ matrix.package }}
+        env:
+          NEO_USER: neo4j
+          NEO_PASSWORD: ${{ secrets[env.AURA_PASSWORD_SECRET] }}
+          NEO_URL: ${{ secrets[env.AURA_URI_SECRET] }}
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          directory: ./packages/${{ matrix.package }}/coverage/
+          flags: ${{ matrix.package }},integration,aura-${{ matrix.aura-instance }}
+          fail_ci_if_error: true
```

Comment:
Should a failed upload to codecov really fail the integration tests?
Also, what's the reason for uploading the code coverage results on each integration test setup?

Reply:
I can change that!

For sure, I could upload each coverage report as an artifact after each test run, and then have an "upload coverage" job which uploads them all - we wouldn't have flags "unit", "TCK", etc. this way, so maybe not so optimal.

Alternative is to upload all integration test reports in one... Let me have a play around. 🙂

Reply:
Cool, integration test coverage reports now upload in one following all on-prem and Aura runs. 😎